sql/core/src/main/scala/org/apache/spark/sql/execution/AggregatingAccumulator.scala

```diff
@@ -95,13 +95,13 @@ class AggregatingAccumulator private(
 
   /**
    * Driver side operations like `merge` and `value` are executed in the DAGScheduler thread. This
-   * thread does not have a SQL configuration so we attach our own here. Note that we can't (and
-   * shouldn't) call `merge` or `value` on an accumulator originating from an executor so we just
-   * return a default value here.
+   * thread does not have a SQL configuration so we attach our own here.
    */
   private[this] def withSQLConf[T](default: => T)(body: => T): T = {
     if (conf != null) {
       SQLConf.withExistingConf(conf)(body)
+    } else if (TaskContext.get != null) {
+      SQLConf.withExistingConf(SQLConf.get)(body)
```

Comment:
This does nothing. `SQLConf.withExistingConf(abc)` makes sure `SQLConf.get` will return `abc` later. We can just write `body`